app/viewers/embed/viewer/common_viewer.rb

```diff
@@ -34,9 +34,12 @@ def external_url
       # Creates a file url for stacks
       # @param [String] title
       # @param [Boolean] download
+      # @param [Boolean] allow_literal_slashes
       # @return [String]
-      def file_url(title, download: false)
-        uri = URI.parse("#{stacks_url}/#{ERB::Util.url_encode(title)}")
+      def file_url(title, download: false, allow_literal_slashes: false)
+        encoded_title = ERB::Util.url_encode(title)
+        encoded_title.gsub!('%2F', '/') if allow_literal_slashes
```

Comment:
I don't know if rubocop would like it, but maybe something like this would be a little more robust?

```suggestion
        parts =  allow_literal_slashes ? title.split('/') : [title]
        encoded_title  = title.map { |x| ERB::Util.url_encode(x) }.join('/')
```

I get nervous about trying to undo encoded stuff after the fact 🤷‍♂️